src/Web/WebClient.php

```diff
@@ -395,19 +395,19 @@ protected function detectEngine($userAgent)
             $this->engine = self::BLINK;
         } elseif (\stripos($userAgent, 'Chrome') !== false) {
             $result  = \explode('/', \stristr($userAgent, 'Chrome'));
-            $version = \explode(' ', $result[1]);
+            $version = isset($result[1]) ? \explode(' ', $result[1]) : false;
 
-            if ($version[0] >= 28) {
+            if ($version === false || $version[0] >= 28) {
                 $this->engine = self::BLINK;
             } else {
                 $this->engine = self::WEBKIT;
             }
         } elseif (\stripos($userAgent, 'AppleWebKit') !== false || \stripos($userAgent, 'blackberry') !== false) {
             if (\stripos($userAgent, 'AppleWebKit') !== false) {
                 $result  = \explode('/', \stristr($userAgent, 'AppleWebKit'));
-                $version = \explode(' ', $result[1]);
+                $version = isset($result[1]) ? \explode(' ', $result[1]) : false;
 
-                if ($version[0] === 537.36) {
+                if ($version !== false && $version[0] === 537.36) {
                     // AppleWebKit/537.36 is Blink engine specific, exception is Blink emulated IEMobile, Trident or Edge
                     $this->engine = self::BLINK;
```

Comment:
Yes ... weird. Would have to check with Git Blame if someone has removed a "return" statement or how else that code was changed so it is that weird.

Reply:
It was added with this PR in 2016: #62 . And that created that nonsense.

Reply:
Move line 417 above the if statement should fix the issue.

Reply:
or put it into an else clause:
```PHP
} elseif (\stripos($userAgent, 'AppleWebKit') !== false || \stripos($userAgent, 'blackberry') !== false) {
  if (\stripos($userAgent, 'AppleWebKit') !== false) {
     ...
  }
  else
  {
    // Evidently blackberry uses WebKit and doesn't necessarily report it.  Bad RIM.
    $this->engine = self::WEBKIT;
  }
```

Reply:
I've refactored the code so it makes sense. It seems the current unit tests do not really cover the case for AppleWebKit/537.36 because they pass regardless if this code is present or not. It also doesn't matter for the existing unit test cases if I check for `$version[0] === 537.36` (which is wrong) or `$version[0] === '537.36'`. I will fix that soon with another commit.